src/gui/pedalboard/cables/CableView.h

```diff
@@ -51,6 +53,36 @@ class CableView : public Component,
     std::unique_ptr<CableViewPortLocationHelper> portLocationHelper;
 
     EditorPort nearestPort {};
+    juce::Point<int> portToPaint;
+
+    bool mouseOverClickablePort();
+    bool mouseDraggingOverOutputPort();
+
+    CriticalSection cableMutex;
+
+    struct PathGeneratorTask : juce::TimeSliceClient
+    {
+    public:
+        explicit PathGeneratorTask (CableView& cv) : cableView (cv)
+        {
+            sharedTimeSliceThread->addTimeSliceClient (this);
+
+            if (! sharedTimeSliceThread->isThreadRunning())
+                sharedTimeSliceThread->startThread();
+        }
+
```

Comment:
We should also have a destructor here that removes the task from the `TimeSliceThread`.

Reply:
Ah right, thanks for catching that